### PR TITLE
trim snapshot progress output

### PR DESCRIPTION
**Pull request by tp-woven**
Following [this discussion](https://discord.com/channels/968932220549103686/969829516539228222/1105304374584025251) on Discord, this trims the output of the snapshot progress messages so that they can be cleared correctly.

### CODE REVIEW

File: src/progress.rs [hunk 1]
```diff
@@ -196,12 +196,15 @@ pub fn snapshot_progress(ui: &mut Ui) -> Option<impl Fn(&RepoPath) + '_> {
...
             );
         }
-        _ = write!(
-            state.ui,
-            "\r{}Snapshotting {}",
-            Clear(ClearType::CurrentLine),
-            path.to_fs_path(Path::new("")).display()
-        );
+
+        let bar_width = state.ui.term_width().map(usize::from).unwrap_or(20);
```
Comment (tp-woven) [inline src/progress.rs:200]: 20 felt like a reasonable value that would capture the "Snapshotting" part and a few characters from the path, but I'm open to suggestions...

File: src/progress.rs [hunk 2]
```diff
@@ -196,12 +196,15 @@ pub fn snapshot_progress(ui: &mut Ui) -> Option<impl Fn(&RepoPath) + '_> {
...
+        let msg: String = format!("Snapshotting {}", path.to_fs_path(Path::new("")).display())
+            .chars()
```
Comment (tp-woven) [inline src/progress.rs:202]: Is `chars()` OK to use here, or should we use grapheme clusters?

Comment (Ralith) [reply src/progress.rs:202]: Rather than mucking with iterators at all, format with `"\r{}{:.*}", Clear(...), line_width, msg` (see [docs](https://doc.rust-lang.org/std/fmt/#precision))

File: src/progress.rs (hunk 1 above)
Comment (Ralith) [reply src/progress.rs]: Six characters of the path seems of pretty limited. use. How about `80`, as a typical terminal width? Note also this path only runs when stdout is a tty so I doubt this will fail often.

File: src/progress.rs (hunk 2 above)
Comment (tp-woven) [reply src/progress.rs:202]: Oh, nice! TIL 😺

File: src/progress.rs (hunk 1 above)
Comment (tp-woven) [reply src/progress.rs]: Done.

File: src/progress.rs (hunk 2 above)
Comment (tp-woven) [reply src/progress.rs]: OK, this looks much nicer (and now only has a single string format).

**Comment by tp-woven** [comment]
@Ralith Could you take another look? 🙏

File: .gitignore [hunk 3]
```diff
@@ -1,6 +1,7 @@
 /target/
 .direnv
 .envrc
+.idea
```
Comment (martinvonz) [inline .gitignore:4]: I think this should go into your personal .gitignore, or at least be sent as a separate PR

**Review by martinvonz — APPROVED** [review]
FWIW, this LGTM.

**Review by Ralith — APPROVED** [review]
LGTM aside from the above. I'm guessing this fixes your issue in practice?

File: .gitignore (hunk 3 above)
Comment (tp-woven) [reply .gitignore:4]: Ack, https://github.com/martinvonz/jj/pull/1598 splits it to another PR.

I can't put it in my personal .gitignore since in some projects I _do_ want to check the folder in - IntelliJ is smart enough to create its own `.idea/.gitignore` to ignore personal settings and only commit "generic" project settings.

**Comment by tp-woven** [comment]
Thanks for the reviews! I'll merge this after #1598, per @martinvonz's comment.

> I'm guessing this fixes your issue in practice?

Yup, I `cargo install`ed this branch locally and it works as intended on my main work repo.

File: .gitignore (hunk 3 above)
Comment (martinvonz) [reply .gitignore:4]: I think we should have a `jj track` for that (i.e. the opposite of `jj untrack`)

Comment (tp-woven) [reply .gitignore]: Hmm, I thought `untrack` was only used to remove files that are `.gitignore`d from the current commit. How would `track` keep track (😅) of what to "unignore" across changes? 🤔

Comment (martinvonz) [reply .gitignore]: I was thinking that `jj track foo` would add that file or directory even if it was in your `.gitignore`
